erpnext/patches.txt

```diff
@@ -291,7 +291,7 @@ erpnext.patches.v7_0.rename_prevdoc_fields
 erpnext.patches.v7_0.rename_time_sheet_doctype
 execute:frappe.delete_doc_if_exists("Report", "Customers Not Buying Since Long Time")
 erpnext.patches.v7_0.make_is_group_fieldtype_as_check
-execute:frappe.reload_doc('projects', 'doctype', 'timesheet') #2016-09-12
+execute:frappe.reload_doc('projects', 'doctype', 'timesheet', force=True) #2016-11-04
```

Comment:
You need to add those permission via a separate patch. If the new role is not exists, add it via patch. But we can't reset the whole permissions, it will remove customization if any.